src/NzbDrone.Core/HealthCheck/Checks/IndexerStatusCheck.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Indexers;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    public class IndexerStatusCheck : HealthCheckBase
+    {
+        private readonly IIndexerFactory _indexerFactory;
+        private readonly IIndexerStatusService _indexerStatusService;
+
+        public IndexerStatusCheck(IIndexerFactory indexerFactory, IIndexerStatusService indexerStatusService)
+        {
+            _indexerFactory = indexerFactory;
+            _indexerStatusService = indexerStatusService;
+        }
+
+        public override HealthCheck Check()
+        {
+            var enabledIndexers = _indexerFactory.GetAvailableProviders();
+            var backOffIndexers = enabledIndexers
+                .Select(v => Tuple.Create(v, _indexerStatusService.GetIndexerStatus(v.Definition.Id)))
```

Comment:
use an anonymous type instead of a tuple. makes things easier to read. also break this in multiple stages with proper variable names.